### ISSUE TITLE
Centralized Documentation Repo/Site

### ISSUE THREAD

**Issue by jhoffner** [Codewars/codewars.com#911]

**Comment by jhoffner** [comment]
@kazk I had it slated for this sprint to focus on this, but I may have to push this off to the first week of August since I'm dealing with a possible CR server refactor (due to some issues that are popping up), and I am mostly out next week for a business trip with limited availability.

I did do a deeper look into the Jekyll implementation and the big question I have is can we implement search functionality and still support automatic deployments? Doing some research it seems Jekyll with Github pages only works if you don't use plugins. 

If it ends up being the case where we can't use Github auto-deployments due to needing additional functionality, then we may want to look into (Metalsmith)[http://www.metalsmith.io/]. 

The main items for me to work through as the next steps are:
- [ ] finalize content structure
- [ ] homepage and article design

**Comment by kazk** [comment]
> can we implement search functionality and still support automatic deployments?

We can, by generating data to be used by client side library.
Current search functionality depends on the data in [docs/assets/data.json.txt](https://github.com/Codewars/codewars.com/blob/ba4c43ee398c874844f3782268074781ff55832f/docs/assets/data.json.txt) which is populated automatically and performed by [lunr.js](https://lunrjs.com/).
I haven't compared the functionality or the efficiency against Jekyll plugins.

> Jekyll with Github pages only works if you don't use plugins.

In the current setup, we can't use Jekyll plugins that's not in the GitHub's whitelist. We can avoid this limitation by setting up some external service to do the automatic deployments though (just use GitHub Pages as hosting service). If we're doing that, using other static site generators like Metalsmith is an option too.

I tried to make things simple because I wasn't sure what we wanted. So existing content should be able to adapt to other tools if we want. I don't think I used any of the whitelisted plugins or any JavaScript except for the search.
